packages/ui/src/composables/useHover.ts

```diff
@@ -1,6 +1,16 @@
-import { ref, watch, onMounted, onBeforeUnmount, Ref } from 'vue'
+import { PropType, ref, Ref, watch } from 'vue'
 
-const getEl = (el: any) => el.$el !== undefined ? el.$el : el
+export const useHoverProps = {
+  hoverBehaviour: {
+    type: String as PropType<'opacity' | 'mask'>,
+    default: 'mask',
+    validator: (value: string) => ['opacity', 'mask'].includes(value),
+  },
+  hoverOpacity: { type: Number, default: 0.15 },
+  hoverMaskColor: { type: String, default: 'white' },
```

Comment:
Can we do 

```
type ColorString = string
```

And then use ColorString here?

But I can create an issue to discuss if you don't find this exciting :D.